### ISSUE TITLE
Advanced rounding: Multiples and non-numeric values

### ISSUE THREAD

**Issue by philrz** [brimdata/super#5442]
## tl;dr

A user asked in a recent [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1730920605821519):

> is there an easy  way to floor a duration? `18m2.43245343s` => `18m2s`

There doesn't seem to be a direct way to achieve this in the language currently, but it certainly seems useful. Excel's [MROUND](https://exceljet.net/functions/mround-function) and related functions seem like a nice template of ideas of how we might address this.

## Details

At the time this issue is being filed, super is at commit. 92e3052.

FWIW, the specific result the user was seeking can be obtained in a hacky way by casting the duration into a string, chopping off the fractional seconds portion, and then casting it back into a duration again, i.e.,

```
$ super -version
Version: v1.18.0-121-g92e30525

$ echo '18m2.43245343s' | super -z -c 'duration(regexp_replace(string(this), /\..*s$/, "s"))' -
18m2s
```

But for the general case, something more flexible is surely justified.

Looking at what else is out there, Excel's [MROUND](https://exceljet.net/functions/mround-function) has at least two improvements I could see adopting here.

1. They accept time-based values, not just pure numbers. Right now our functions like `round` and `floor` only take numeric arguments, so for a duration value like the user's the functions currently just return the same value unmodified.

2. They take an argument that can specify an arbitrary multiple by which to round/floor. Right now our functions only round/floor/etc. to the nearest integer, but such an argument could allow for things like "round to the nearest 15 minutes".
